tools/bazel/compile_commands/compile_commands.py

```diff
@@ -0,0 +1,255 @@
+                        default="compile_commands.json",
+                        help="output compile_commands.json file")
+    parser.add_argument("-b", "--build",
+                        help="bazel build command arguments")
+    parser.add_argument("-v", "--verbosity",
+                        default=0,
+                        action="count",
+                        help="increase output verbosity (e.g., -v or -vv)")
+    parser.add_argument("--log-format",
+                        default="[bazel] %(levelname)5s: %(message)s",
+                        help=argparse.SUPPRESS)
+
+    options = parser.parse_args()
+
+    if options.verbosity >= 2:
+        log_level = logging.DEBUG
+    elif options.verbosity >= 1:
+        log_level = logging.INFO
+    else:
+        log_level = logging.WARN
+    logging.basicConfig(level=log_level, format=options.log_format)
+
+    return options
+
+
+def split_to_list(arguments):
+    """
+    Split argument string to list
+    """
+    if type(arguments) is list:
+        return arguments
+    else:
+        return shlex.split(arguments)
+
+
+def run_command(command, stdout=subprocess.PIPE, stderr=subprocess.PIPE):
+    """
+    Run shell command
+    """
+    cmd = split_to_list(command)
+    process = subprocess.Popen(cmd, stdout=stdout, stderr=stderr)
+    out, err = process.communicate()
+    if process.returncode:
+        logging.error("Command: %s...\nError: %s", command, err)
+    return out.decode("utf-8")
+
+
+def bazel_aquery(build_arguments):
+    """
+    Run bazel query and return list of source files, e.g.:
+    bazel cquery 'kind("source file", deps("@workspace//target"))' --config=something
+        --noimplicit_deps --notool_deps
+    """
+    arguments = split_to_list(build_arguments)
+    targets = []
+    configs = []
+    bazel = arguments[0]
+    logging.debug("Bazel: %s", bazel)
+    logging.debug("Bazel build arguments:")
+    for argument in arguments:
+        logging.debug("   %s", argument)
+        if re.search(r"^(@|:|\/\/)\S*", argument) or argument == "...":
+            targets.append(argument)
+        elif re.search(r"^--\S*", argument):
+            configs.append(argument)
+    logging.debug("Targets: %s", targets)
+    logging.debug("Configs: %s", configs)
+    if len(targets) < 1:
+        logging.error("No targets found for bazel query in: %s", build_arguments)
+        return
+
+    deps = " + ".join(["deps(\"{target}\")".format(target=target) for target in targets])
+    logging.debug("Deps: %s", deps)
+
+    options = [
+        "--output=jsonproto",
+        "--include_artifacts=false",
+    ]
+    command = "{bazel} aquery {options} 'mnemonic(\"CppCompile\", {deps})' {configs}".format(
+        bazel=bazel, deps=deps, options=" ".join(options), configs=" ".join(configs))
+    logging.debug("Command: %s", command)
+
+    out = run_command(command)
+    if out:
+        data = json.loads(out)
+        return data
+    else:
+        return False
```

Comment:
Now the function may return with `None` (see the code above L101), a dictionary or a Boolean (False) value. Couldn't we return here with None if the output is empty?
```suggestion
    if out:
        return json.loads(out)
```

Reply:
Done